File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def height(root):
            if not root:
                return 0
            left = height(root.left)
            if left == -1:
                return -1
            right = height(root.right)
            if right == -1:
                return -1

            abso = abs(left - right)
            if abso > 1:
                return -1
            
            return 1 + max(left, right)
        return height(root) != -1
        

File: Data Structures & Algorithms/path-sum/submission-0.cpp
class Solution {
public:
    bool hasPathSum(TreeNode* root, int targetSum) {
        if (!root)
            return false;
        targetSum -= root->val;
        if(!root->left && !root->right){
            return targetSum == 0; 
        }

        bool left = hasPathSum(root->left, targetSum);
        bool right = hasPathSum(root->right, targetSum);


        
        return left || right;

    }
};